### PR TITLE
Pull request for offline cache async operations

### PR DESCRIPTION
**Pull request by ashokkumarprajapati**
Updated code to use async cache Operations rather than synchronous.

### CODE REVIEW

File: components/calExchangeCalendar.js [hunk 1]
```diff
@@ -8343,37 +8390,37 @@ this.logInfo("getTaskItemsOK 4");
...
+      				 }
+     				});
 			var sqlStr = "INSERT INTO attachments VALUES ('"+attParams.id+"', '"+attParams.name.replace(/\x27/g, "''")+"', '"+attParams.size+"', '')";
-			if (!this.executeQuery(sqlStr)) {
-				this.logInfo("Error inserting attachment into offlineCacheDB. Error:("+this.offlineCacheDB.lastError+")"+this.offlineCacheDB.lastErrorString);
-			}
-			else {
-				this.logInfo("Inserted attachment into offlineCacheDB. Name:"+attParams.name+", Title:"+aCalItem.title);
-			}
+			this.executeQueryAsync(sqlStr);
```
Comment (1stsetup) [inline components/calExchangeCalendar.js]: Please add success and error messages to the calls to executeQueryAsync.

File: components/calExchangeCalendar.js [hunk 2]
```diff
@@ -8278,6 +8306,24 @@ this.logInfo("getTaskItemsOK 4");
 		}
 	},
 
+	executeQueryAsync: function _executeQueryAsync(aQuery,successMsg,errorMsg)
+	{
+		this.logInfo("sql-query:"+aQuery, 2);
+		try {
+			var sqlStatement = this.offlineCacheDB.createStatement(aQuery);
+			sqlStatement.executeAsync();
+			sqlStatement.reset();
```
Comment (1stsetup) [inline components/calExchangeCalendar.js]: This is dangerous because the reset could be done before the executeAsync is done. Please add the handlers (error and completion) to the executeAsync. Remove the reset.

File: components/calExchangeCalendar.js [hunk 3]
```diff
@@ -8556,13 +8568,7 @@ this.logInfo("getTaskItemsOK 4");
...
 			this.logInfo("sql-query:"+sqlStr, 2);
 			var sqlStatement = this.offlineCacheDB.createStatement(sqlStr);
-			sqlStatement.executeStep();
-			if (sqlStatement.row.itemcount > 0) {
-				this.logInfo("This item is allready in offlineCache database. id and changeKey are the same. Not going to update it.");
-				sqlStatement.reset();
-				return;
-			}
-			sqlStatement.reset();
+			sqlStatement.execute();
```
Comment (1stsetup) [inline components/calExchangeCalendar.js]: Why was this part changed? Now you are going to update items which are allready in the offline cache. Or am I missing something.

File: chrome/content/attachments-view.xul [hunk 4]
```diff
@@ -87,8 +89,17 @@
 			disable-on-readonly="true"
 			removable="true"
 			command="exchWebService_addAttachmentDialog"
+			oncommand="exchWebService.attachments.addAttachmentDialog();" />
+		<toolbarbutton id="button-url"
```
Comment (1stsetup) [inline chrome/content/attachments-view.xul]: Please create a seperate pull request for this part. It is not part of the Async functionality of the rest.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Exchnage 2007/2010 Calendar and task provider#
```
Comment (1stsetup) [inline README.md]: Please create seperate pull request for this new file. It is not part of the Async functionality.

File: components/calExchangeCalendar.js (hunk 2 above)
Comment (ashokkumarprajapati) [inline components/calExchangeCalendar.js]: Done. Commented reset statement.

File: components/calExchangeCalendar.js (hunk 3 above)
Comment (ashokkumarprajapati) [inline components/calExchangeCalendar.js]: My mistake.  updated it back to original.

File: components/calExchangeCalendar.js (hunk 1 above)
Comment (ashokkumarprajapati) [inline components/calExchangeCalendar.js]: Added success/error messages to calls to executeQueryAsync.

File: README.md (hunk 5 above)
Comment (ashokkumarprajapati) [inline README.md]: Done. Removed readme.

**Comment by ashokkumarprajapati** [comment]
Should we go ahead and merge these changes now?

**Comment by 1stsetup** [comment]
Ashok, I did not yet have time too look at it or test it.

**Comment by ashokkumarprajapati** [comment]
no problem. Let me know when done.

**Comment by ashokkumarprajapati** [comment]
Michel, did you got chance to look at this?

**Comment by 1stsetup** [comment]
Yes but I cannot merge it anymore into the latest version with has the e4xReplacement in it.
So it will be in a next version.

**Comment by ashokkumarprajapati** [comment]
Any update on this?
